src/hip_hcc.cpp

```diff
@@ -791,6 +791,7 @@ hipError_t ihipDevice_t::initProperties(hipDeviceProp_t* prop) {
     err = hsa_agent_get_info(_hsaAgent, HSA_AGENT_INFO_NAME, &archName);
 
     prop->gcnArch = atoi(archName + 3);
```

Comment:
Arch may be hexidecimals, so atoi will not work.

Reply:
gcnArch now represents a series of arch's with the help of major number. For all gfx 9 series, gcnArch will return 900 etc. For getting the exact ISA, one has to use gcnArchName as it represents full name. eg: gfx906, gfx90a etc.